### PR TITLE
Analysis of flood waves

### PR DESCRIPTION
**Pull request by vmarsi**

### CODE REVIEW

**Comment by zsvizi** [comment]
We need conceptual changes:
- `flood_wave_detector.py` technically speaking does not provide *flood waves* - it find the nodes and the edges for the flood graph -> a better naming is needed, e.g. `GraphPreparator` (I know, bad) - consequently, `FloodWaveHandler` has to be `GraphHandler` instead.
- the following methods are responsible for flood wave extraction: `GraphAnalysis.get_flood_waves`,  `GraphAnalysis.get_flood_waves_without_equivalence`, `GraphAnalysis.get_full_flood_waves` -> these should go into a `FloodWaveDetector` or `FloodWaveExtractor` class, which should be in the module `core` - this class has not to be a static class, thus `joined_graph` and `waves` could be saved in member variables.
- The method `GraphAnalysis.create_flood_map` might need an own class with name `FloodMapCreator`.
- The method `AnalysisHandler.get_final_pairs` should go to `FloodWaveHandler` (a renewed one, since the previous content goes to a different class).
- My feeling is, that `SelectionHandler.select_full_from_start_to_end` should also go to `FloodWaveHandler`.

File: src/analysis/graph_analysis.py [hunk 1]
```diff
@@ -16,277 +17,258 @@ class GraphAnalysis:
...
+                end_node = paths[0][-1]
+                amount = len(paths)
+
+                edges.append((start_node, end_node, amount))
+
+        flood_map.add_weighted_edges_from(ebunch_to_add=edges)
+
         return flood_map
 
     @staticmethod
```
Comment (zsvizi) [inline src/analysis/graph_analysis.py]: I do not see that you use this method anywhere.

File: src/analysis/graph_analysis.py [hunk 2]
```diff
@@ -315,19 +297,21 @@ def get_branching(joined_graph: nx.DiGraph) -> list:
         return final_comps
 
     @staticmethod
-    def calculate_all_velocities(joined_graph: nx.DiGraph, river_kms: pd.Series) -> list:
+    def calculate_all_velocities(joined_graph: nx.DiGraph) -> list:
```
Comment (zsvizi) [inline src/analysis/graph_analysis.py]: I would put this method to `statistical_analysis.py`.

**Comment by zsvizi** [comment]
Could you set up a Jupyter notebook containing all possible analysis pieces? Additionally, we need a block for visualizing the distribution of slopes for a chosen pair of stations and for a chosen interval.

File: src/analysis/graph_analysis.py [hunk 3]
```diff
@@ -16,277 +17,258 @@ class GraphAnalysis:
...
-        :param Callable func:
-        :return list: The results organized into a list.
+        waves = []
+        for comp in components:
+            final_pairs = AnalysisHandler.get_final_pairs(joined_graph=joined_graph, comp=list(comp))
+
+            for start, end in final_pairs:
+                try:
+                    wave = nx.shortest_path(joined_graph, start, end)
+                    waves.append(wave)
```
Comment (zsvizi) [inline src/analysis/graph_analysis.py]: It is enough to store only the starting and end nodes instead of the path.

**Comment by vmarsi** [comment]
Gdown javítása

**Review by zsvizi — APPROVED** [review]

**Pull request merged**